### ISSUE TITLE
duplicate_non_manifold_vertices causes isotropic_remeshing to never return 

### ISSUE THREAD

**Issue by jasjuang** [CGAL/cgal#4133]
## Issue Details

For this specific mesh [test1.zip](https://github.com/CGAL/cgal/files/3462248/test1.zip) I worked with, if I run `duplicate_non_manifold_vertices` first then `isotropic_remeshing` will be stuck somewhere internally and never return. The reason I ran `duplicate_non_manifold_vertices` is because in a previous issued I filed #3565, running `duplicate_non_manifold_vertices` first actually is required in order for `isotropic_remeshing` to work. The dilemma for me is I can't have different codes for different meshes. Ideally, I need to decide to add `duplicate_non_manifold_vertices` or not, and then `isotropic_remeshing` should work for all kinds of meshes. Please see below minimal code to reproduce the problem

```
typedef CGAL::Simple_cartesian<double> K;
typedef CGAL::Surface_mesh<K::Point_3> Mesh;

Mesh m;

std::ifstream input("test1.off");
input >> m;
input.close();

// false
std::cout << CGAL::is_valid_polygon_mesh(m) << std::endl; 

CGAL::Polygon_mesh_processing::duplicate_non_manifold_vertices(m);

// false
std::cout << CGAL::is_valid_polygon_mesh(m) << std::endl;

// never returns
CGAL::Polygon_mesh_processing::isotropic_remeshing(faces(m), 5.0, m)
```

## Environment

* Operating system: Ubuntu 18.04
* Compiler: GCC 7
* Release or debug mode: Release
* CGAL version: Latest master
* Boost version: 1.68

**Comment by MaelRL** [comment]
There was indeed some issues with the manifold vertex duplication function. This mesh really was bending the rules (e.g. the same vertex being used in two stars while being technically a single flat star, etc.), so it was interesting to improve the function, thanks! Do you mind / is it possible from a license point of view to please use it in the CGAL testsuite? Otherwise, I will just extract the local combinatorics and change the geometry, but it's tedious :).

Things should be fixed with https://github.com/MaelRL/cgal/commit/814f8c67e56297167e9a6f57d90fa166fabf5a4a, but I worked in an unrelated branch and I have to backport things. Isotropic remeshing runs fine afterwards. [Here](https://gist.github.com/MaelRL/961367357931dcfd1aec203e24af43d8) is what I get, in case you want to already progress with it.

Note that you can also call `remove_isolated_vertices()` initially, you were carrying about a hundred useless points.

**Comment by jasjuang** [comment]
@MaelRL No problem feel free to add the mesh into the CGAL testsuite. Is it possible for me to directly apply the change on your fork manually to the lastest CGAL master branch to test? Or is it better for me to wait for a pullrequest?

**Comment by MaelRL** [comment]
See https://github.com/CGAL/cgal/pull/4139 (pending tests)

**Comment by jasjuang** [comment]
Is #4137 required also in order for me to test? If that's the case, when I checkout #4139 and then merge #4137 into it there's a conflict. Or do I just need #4139?

**Comment by MaelRL** [comment]
If your branch is master, #4139 is sufficient. #4137 is just the same bug fix, back ported for another bug fix release (4.14.1). There are different PRs because there were some significant changes to this function since release 4.14 (but the bug was still present for both)?

**Comment by jasjuang** [comment]
Thanks for the clarification. I tested #4139 and I can confirm that it solves the problem for me. Please merge!
